pkg/controller/node/nodecontroller.go

```diff
@@ -189,6 +191,15 @@ func NewNodeController(
 			UpdateFunc: func(_, obj interface{}) { nc.maybeDeleteTerminatingPod(obj) },
 		},
 	)
+
+	nodeEventHandlerFuncs := framework.ResourceEventHandlerFuncs{}
+	if nc.allocateNodeCIDRs {
+		nodeEventHandlerFuncs = framework.ResourceEventHandlerFuncs{
```

Comment:
I'm not very familiar with the ResourceEventHandlers, but I don't think this would handle crashes/restarts well.

E.g., if the nodeController restarts, it is possible that the next new node is issued a CIDR that was previously issued to someone else, right?